packages/playwright/src/generators/configuration/generator.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  addDependenciesToPackageJson,
+  convertNxGenerator,
+  formatFiles,
+  generateFiles,
+  GeneratorCallback,
+  readNxJson,
+  readProjectConfiguration,
+  runTasksInSerial,
+  toJS,
+  Tree,
+  updateNxJson,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+import * as path from 'path';
+import { ConfigurationGeneratorSchema } from './schema';
+import { nxVersion, playwrightVersion } from '../../utils/versions';
+
+export async function configurationGenerator(
+  tree: Tree,
+  options: ConfigurationGeneratorSchema
+) {
+  const tasks: GeneratorCallback[] = [];
+
+  if (!options.skipPackageJson) {
+    tasks.push(
+      addDependenciesToPackageJson(
+        tree,
+        {},
+        {
+          '@nx/playwright': nxVersion,
+          '@playwright/test': playwrightVersion,
+        }
+      )
+    );
+  }
+
+  const projectConfig = readProjectConfiguration(tree, options.project);
+  generateFiles(
+    tree,
+    path.join(__dirname, 'files'),
+    projectConfig.root,
+    options
+  );
+
+  addE2eTarget(tree, options);
+  setupE2ETargetDefaults(tree);
+  addGitIgnore(tree);
+
+  if (options.js) {
+    toJS(tree);
+  }
+  if (!options.skipFormat) {
+    await formatFiles(tree);
+  }
+
+  return runTasksInSerial(...tasks);
+}
+
+function setupE2ETargetDefaults(tree: Tree) {
+  const nxJson = readNxJson(tree);
+
+  if (!nxJson.namedInputs) {
+    return;
+  }
+
+  // E2e targets depend on all their project's sources + production sources of dependencies
+  nxJson.targetDefaults ??= {};
+
+  const productionFileSet = !!nxJson.namedInputs?.production;
+  nxJson.targetDefaults.e2e ??= {};
+  nxJson.targetDefaults.e2e.inputs ??= [
+    'default',
+    productionFileSet ? '^production' : '^default',
+  ];
+
+  updateNxJson(tree, nxJson);
+}
+
+function addE2eTarget(tree: Tree, options: ConfigurationGeneratorSchema) {
+  const projectConfig = readProjectConfiguration(tree, options.project);
+  if (projectConfig?.targets?.e2e) {
+    throw new Error(`Project ${options.project} already has an e2e target.
+Rename or remove the existing e2e target.`);
+  }
+  projectConfig.targets.e2e = {
+    executor: '@nx/playwright:playwright',
+    options: {},
+  };
+  updateProjectConfiguration(tree, options.project, projectConfig);
+}
+
+function addGitIgnore(tree: Tree) {
+  const contents = tree.read('.gitignore', 'utf-8') ?? '';
+  if (/playwright/gm.test(contents)) {
+    return;
+  }
+
+  tree.write(
+    '.gitignore',
+    [
+      contents,
+      '/test-results/',
+      '/playwright-report/',
+      '/playwright/.cache/',
```

Comment:
we probably don't need to add these as, we should aim to put the output of playwright (screenshots/videos/reports) in the `dist/playwright/<project-root>` directory like we do for cypress. 
1. to keep consistency between cypress and playwright
2. to prevent each project from overriding each others artifacts

Reply:
removed this function